### PR TITLE
Add github-ci.yml to set-up ci on Github action

### PR DESCRIPTION
**Pull request by seraphinandrieux**
I created github-ci.yml to implement github action.
I keep the same java (8) and node (10.16.3) version for this CI.

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
LGTM
The only way to really test it, is to merge into master. Let's do it !

File: .github/workflows/github-ci.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Application CI
+on: [push, pull_request]
+jobs:
+    pipeline:
+        name: jhipster-online pipeline
+        runs-on: ubuntu-latest
+        if: "!contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.pull_request.title, '[skip ci]') && !contains(github.pull_request.title, '[ci skip]')"
```
Comment (SudharakaP) [inline .github/workflows/github-ci.yml]: When taking the pull request title; it should be; `github.event.pull_request.title`. We have changed this on the [other Github Actions](https://github.com/jhipster/generator-jhipster/blob/ec26d48bf7683ba768d4ff07dcfcf73e8f141fb1/.github/workflows/react.yml#L40) 😄

File: .github/workflows/github-ci.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        timeout-minutes: 40
+        env:
+            NODE_VERSION: 10.16.3
+            SPRING_OUTPUT_ANSI_ENABLED: DETECT
+            SPRING_JPA_SHOW_SQL: false
+            JHI_DISABLE_WEBPACK_LOGS: true
+            NG_CLI_ANALYTICS: false
+
+        steps:
+            - uses: actions/checkout@v1
```
Comment (SudharakaP) [inline .github/workflows/github-ci.yml]: Perhaps use `actions/checkout version 2`; https://github.com/actions/checkout/releases?

File: .github/workflows/github-ci.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 10.16.3
```
Comment (SudharakaP) [inline .github/workflows/github-ci.yml]: Maybe use node version 12.16.0 for consistency as we use that in [all scripts in the main generator](https://github.com/jhipster/generator-jhipster/blob/ec26d48bf7683ba768d4ff07dcfcf73e8f141fb1/.github/workflows/generator.yml#L31)? 😄

**Comment by pascalgrimaud** [comment]
Sorry, I was too fast.
You're right @SudharakaP :p

File: .github/workflows/github-ci.yml (hunk 3 above)
Comment (SudharakaP) [inline .github/workflows/github-ci.yml]: Do we have to use 10.16.3? In the main generator we use 12.16.0? 🤔

**Comment by SudharakaP** [comment]
@pascalgrimaud : Oh, no problem. Just some things for consideration; maybe could be done in another PR. 😄

**Pull request merged**